lib/skate/detours/detours.ex

```diff
@@ -5,23 +5,121 @@ defmodule Skate.Detours.Detours do
 
   import Ecto.Query, warn: false
   alias Skate.Repo
-
   alias Skate.Detours.Db.Detour
   alias Skate.Settings.User
 
+  @type t :: %__MODULE__{
+          route: String.t(),
+          direction: String.t(),
+          name: String.t(),
+          intersection: String.t(),
+          updated_at: integer(),
+          author_id: integer(),
+          status: :active | :draft | :past
+        }
+
+  @derive Jason.Encoder
+
+  defstruct [
+    :route,
+    :direction,
+    :name,
+    :intersection,
+    :updated_at,
+    :author_id,
+    :status
+  ]
+
   @doc """
   Returns the list of detours.
 
   ## Examples
 
       iex> list_detours()
       [%Detour{}, ...]
-
   """
   def list_detours do
     Repo.all(Detour)
   end
 
+  @doc """
+  Returns the detours grouped by active, draft, and past.
+
+  ## Examples
+
+      iex> grouped_detours()
+      %{
+        active: [%Detour{}, ...],
+        draft: nil,
+        past: [%Detour{}, ...]
+      }
+  """
+  def grouped_detours(user_id) do
+    detours =
+      Detour
+      |> Repo.all()
+      |> Enum.map(fn detour ->
+        try do
+          db_detour_to_detour(detour, user_id)
+        rescue
+          e in FunctionClauseError ->
```

Comment:
Nit (non-blocking): What if instead of rescuing an exception here, we added another pattern-match-y definition of `db_detour_to_detour/2` like so:

```elixir
db_detour_to_detour(invalid_detour) do
  Sentry.capture_message("Invalid detour", data: %{detour: invalid_detour})

  nil
end
```

(I made up the `data:` key being the way to communicate extra data to Sentry - we should double-check that if we go this route).

Happy to do this as a follow-up PR instead (if we even want to do this).

Reply:
Why switch to capture_message?

Reply:
The approach I was suggesting doesn't have an exception to capture. Maybe `capture_exception` would still work though?

Reply:
Ohhhhh I see. What's the improvement by switching to this approach?

Reply:
I think it's more idiomatic Elixir to rely on pattern-matching, rather than an exception.

Also it's more efficient (I don't think we're worried about performance in this case, but exceptions are slow).

Reply:
Leaving as-is. I cannot easily find documentation about the accepted fields for `captureMessage`